lib/medtronic/processData.js

```diff
@@ -638,19 +615,170 @@ var buildBasalRecords = function (records) {
   return postrecords;
 };
 
+function buildPrimeRecords(records) {
+
+  var primeRecords = filterHistory([RECORD_TYPES.PRIME], records);
+  var postrecords = [];
+
+  primeRecords.forEach(function (primedatum) {
+
+    console.log('prime:',primedatum.jsDate, common.bytes2hex(primedatum.head), common.bytes2hex(primedatum.date), common.bytes2hex(primedatum.body));
+
+    var amount = struct.extractByte(primedatum.head,4) / 10.0;
+    var fixed  = struct.extractByte(primedatum.head,2) / 10.0;
+
+    var type = fixed === 0 ? 'tubing' : 'cannula';
+
+    var deviceEvent = cfg.builder.makeDeviceEventPrime()
+      .with_deviceTime(sundial.formatDeviceTime(primedatum.jsDate))
+      .with_primeTarget(type)
+      .with_volume(amount)
+      .set('index', primedatum.index);
+
+    if(type === 'cannula' && fixed !== amount) {
+      // it's possible to cancel a cannula prime,
```

Comment:
Huh. I wonder if that means our data model should be expanded to have an `expectedVolume` for primes... Of dubious clinical relevance, however...